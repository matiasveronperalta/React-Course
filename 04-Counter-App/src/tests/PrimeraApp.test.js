import React from 'react';
import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';


describe('test <PrimeraApp/>', () => {

    test('should be show Hola soy Goku', () => {
        
        const saludo = 'Hola soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        

        expect(wrapper).toMatchSnapshot();
    });

    test('should be show subtitle reciving from props', () => {
        
        const saludo = 'Hola soy Goku';
        const subTitle = 'Hola soy el pillin';
        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo = {subTitle}
            />
        );
        
        const parrafText = wrapper.find('p').text();

        expect(parrafText).toBe(subTitle);
    })
    



    

});
