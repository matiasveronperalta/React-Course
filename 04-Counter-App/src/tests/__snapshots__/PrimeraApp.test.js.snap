// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <PrimeraApp/> should be show Hola soy Goku 1`] = `
<Fragment>
  <h1>
     
    Hola soy Goku
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
